File: frontend/src/services/api/profileService.js
import apiClient from './apiClient';

export const profileService = {
  getProfile: async () => {
    const response = await apiClient.get('/profile');
    return response.data;
  },

  updateProfile: async (data) => {
    const response = await apiClient.put('/profile', data);
    return response.data;
  },

  changePassword: async (data) => {
    const response = await apiClient.put('/profile/change-password', data);
    return response.data;
  },
};

File: backend/src/main/java/com/medistock/profile/service/ProfileService.java
package com.medistock.profile.service;

import com.medistock.profile.dto.request.ChangePasswordRequest;
import com.medistock.profile.dto.request.UpdateProfileRequest;
import com.medistock.profile.dto.response.UserProfileResponse;

public interface ProfileService {
    UserProfileResponse getProfile(String email);
    UserProfileResponse updateProfile(String email, UpdateProfileRequest request);
    void changePassword(String email, ChangePasswordRequest request);
}
